src/app/dev/platforms/desktop/DevToys.Linux/Core/ThemeListener.cs

```diff
@@ -142,14 +136,19 @@ private bool GetBestValueForCompactMode()
 
     private void UpdateSystemSettingsAndApplyTheme()
     {
-        IsHighContrast = _adwStyleManager.HighContrast;
+        IsHighContrast
+            = _gtkSettings.GtkThemeName is not null
+              && _gtkSettings.GtkThemeName.Contains("high", StringComparison.OrdinalIgnoreCase)
+              && _gtkSettings.GtkThemeName.Contains("contrast", StringComparison.OrdinalIgnoreCase);
         CurrentSystemTheme = GetCurrentSystemTheme();
 
         ApplyDesiredColorTheme();
     }
 
     private AvailableApplicationTheme GetCurrentSystemTheme()
     {
-        return _gtkSettings.GtkApplicationPreferDarkTheme ? AvailableApplicationTheme.Dark : AvailableApplicationTheme.Light;
+        return _gtkSettings.GtkApplicationPreferDarkTheme || (_gtkSettings.GtkThemeName?.Contains("Dark", StringComparison.OrdinalIgnoreCase) ?? false)
```

Comment:
Hi @badcel ,
Do you think this is a reliable way to detect the dark theme?

Reply:
I don't have detailed GTK knowledge.

You probably need to check what libadwaita is doing to find the system preferences for dark theme.

I found this by searching the internet: https://github.com/albertosottile/darkdetect/blob/master/darkdetect/_linux_detect.py

But perhaps there are some cross platform APIs : https://docs.flatpak.org/en/latest/portal-api-reference.html#

You would access them probably via DBus. But I find nothing in regard to dark theme. Perhaps it is via the settings? Sorry this is really not my area of expertise. Perhaps you can ask some libadwaita / GTK developer?

Reply:
It might even be that your Linux mint is to old depending on which Ubuntu it is based. I think those things started to work in the last 2 years but don't quote me on it.

Reply:
Thanks :) This is helpful!
I looked into it and it seems like it's kind of good enough. I will start with this at least.